File: 백준/Silver/10828. 스택/스택.py
import sys
def stack(word, stack=[]):
    order=word[0]
    if(order=="push"):
        stack.append(word[1])
    elif(order=="pop"):
        if(len(stack)!=0):
            
            print(stack.pop())
        else:
            print(-1)
    elif(order=="size"):
        print(len(stack))
    elif(order=="empty"):
        if(len(stack)==0):
            print(1)
        else:
            print(0)
    elif(order=="top"):
        if(len(stack)==0):
            print(-1)
        else:
            print(stack[-1])





        



t=int(sys.stdin.readline())
for i in range(t):
    word=sys.stdin.readline().split()
    stack(word)

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

class Main{
    public static  void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        StringTokenizer st;
        for(int i=0; i<n; i++){
            st = new StringTokenizer(br.readLine());

            String s = st.nextToken();
            if(s.equals("push")){
                int a = Integer.parseInt(st.nextToken());
                stack.add(a);
            }

            if(s.equals("pop")){
                if(!stack.isEmpty()){
                    System.out.println(stack.pop());
                }else{
                    System.out.println(-1);
                }
            }

            if(s.equals("size")){
                System.out.println(stack.size());
            }

            if(s.equals("empty")){
                if(stack.isEmpty()){
                    System.out.println(1);
                }else {
                    System.out.println(0);
                }

            }

            if(s.equals("top")){
                if(stack.isEmpty()){
                    System.out.println(-1);
                }else{
                    System.out.println(stack.peek());
                }
            }
        }

    }
}